### ISSUE TITLE
When using v2.1.3 and v2.1.4, after editing a file and clicking save, the plugin gets stuck. About a 25% probability

### ISSUE THREAD

**Issue by NyxJae** [cline/cline#699]
### Which API Provider are you using?

OpenAI Compatible

### Which Model are you using?

Claude 3.5 Sonnet

### What happened?

![PixPin_2024-11-08_11-38-36](https://github.com/user-attachments/assets/12de4ec6-5875-454a-9af2-78d051193476)
![image](https://github.com/user-attachments/assets/3a688d28-8da4-430c-a9a7-9bd00fd31860)
It won't happen when reverting to v2.1.1.

### Steps to reproduce

Just wait for it to finish writing the file, then click save
After that, you can only restart VSCode or disable the plugin

**Comment by NyxJae** [comment]
Saw similar issues #694 #644

**Comment by NyxJae** [comment]
![image](https://github.com/user-attachments/assets/716c3091-6ac2-435f-9239-db6ff66c736c)

**Comment by NyxJae** [comment]
After this problem recurs, it will be fine for a while after killing all processes of vscode and restarting

**Comment by QDaed** [comment]
i'm having the same and need to resume it then prompting again

**Comment by volome** [comment]
Yes I have the same problem, its ok to restart but to not be able to get back to the last task after restart is crazy

**Comment by dcbartlett** [comment]
Is this still an issue with the latest version?
